src/prng/mod.rs

```diff
@@ -59,82 +59,72 @@
+//! and CSPRNGs).
 //!
 //! CSPRNGs are a little different in that they typically generate a block of
 //! output in a cache, and pull outputs from the cache. This allows them to have
 //! good amortised performance, and reduces or completely removes the influence
-//! of surrounding code on the CSPRNG performance.
-//!
-//! ## Worst-case performance
-//!
-//! Simple PRNGs tend to use constant-time functions and thus have very good
-//! worst case performance. In contrast, CSPRNGs usually produce a block of
-//! values into a cache, giving them poor worst case performance.
+//! of surrounding code on the CSPRNG performance. It does however cause *poor
+//! worst case performance*.
 //!
 //! ## State size
 //!
-//! Simple PRNGs often use very little memory, from as little as one word but
-//! commonly 2-4 words, where a *word* is usually either `u32` or `u64`. This
-//! is not universal across non-cryptographic PRNGs however; for example the
-//! previously popular Mersenne Twister MT19937 algorithm requires 2.5 kB of
-//! state.
+//! Normal PRNGs often use very little memory, commonly only a few words, where
+//! a *word* is usually either `u32` or `u64`. This is not universal however,
+//! for example the historically popular Mersenne Twister MT19937 algorithm
+//! requires 2.5 kB of state.
 //!
 //! CSPRNGs typically require more memory; since the seed size is recommended
-//! to be at least 192 bits and more memory is required for a counter, 256 bits
-//! would be approximately the minimum secure size. In practice, CSPRNGs tend
-//! to use a lot of memory; [`ChaChaRng`] is relatively small with 512 bits
-//! state + 512 bits cache + cache counter (136 bytes on x86_64) while [`Hc128Rng`]
-//! requires over 4 kB.
+//! to be at least 192 bits and some more may be required for the algorithm,
+//! 256 bits would be approximately the minimum secure size. In practice,
+//! CSPRNGs tend to use quite a bit more, [`ChaChaRng`] is relatively small with
+//! 136 bytes.
 //! 
-//! ## Initialisation time
+//! ## Initialization time
 //!
-//! The time required to initialise new generators varies significantly. Many
+//! The time required to initialize new generators varies significantly. Many
 //! simple PRNGs and even some cryptographic ones (including [`ChaChaRng`])
 //! only need to copy the seed value and some constants into their state, and
 //! thus can be constructed very quickly. In contrast, CSPRNGs with large state
-//! require key-expansion, often achieved by using their state-advance function
-//! many times over.
+//! require an expensive key-expansion.
 //!
 //! # Quality
 //!
-//! PRNGs are based in mathematical Number Theory. The quality of PRNG
-//! algorithms can be evaluated both analytically and empirically in order to
-//! examine for bias, correlations and cycle length.
-//!
-//! Many simple PRNGs are not much more than a couple of bitwise and arithmetic
+//! Many normal PRNGs are not much more than a couple of bitwise and arithmetic
```

Comment:
simple was intended here: not all non-csprngs are simple. But this also works, except I am *still* not happy with "normal" to mean non-crypto.